source/javascript/src/collection.js

```diff
@@ -1,16 +1,81 @@
-import { CollectionItem } from './components/CollectionItem.js'
-import { IndexedDBWrapper } from './indexedDB/IndexedDBWrapper.js'
+import { CollectionItem, wrapper } from './components/CollectionItem.js'
 
-// const RealAddButton = document.getElementById('add')
-// const CustAddButton = document.getElementsByClassName('custAdd')
+const addBtn = document.getElementById('add')
 
-// CustAddButton.addEventListener('click', () => {
-//   RealAddButton.click()
-// })
+addBtn.addEventListener('click', () => {
+  // location.pathname = '/source/html/collection-edit.html'
+  const collectionName = window.prompt('Please enter the name of your new collection:')
+  addCollection(collectionName)
+})
+
+/**
+ * Add new collection to database with given name.
+ * @author William Chung <wchung@ucsd.edu>
+ * @param name Name to be given to new collection.
+ */
+function addCollection (collectionName) {
+  // if user presses 'Cancel' on prompt
+  if (collectionName === null) {
+    return
+  }
+
+  const newCollection = new CollectionItem()
+  newCollection.entry = { name: collectionName }
+
+  wrapper.transaction((event) => {
+    const db = event.target.result
+
+    const transaction = db.transaction(['currentLogStore'], 'readwrite')
+    const objectStore = transaction.objectStore('currentLogStore')
+    objectStore.openCursor().onsuccess = function (event) {
+      const cursor = event.target.result
+      if (cursor) {
+        const json = cursor.value
+
+        const collections = json.properties.collections
+
+        const newCollectionObj = {
+          type: 'array',
+          name: collectionName,
+          tasks: [
+          ],
+          images: [
+          ],
+          videos: [
+          ]
+        }
+        collections.push(newCollectionObj)
 
-// RealAddButton.addEventListener('click', () => {
-//   location.pathname = '/source/html/collection-edit.html'
-// })
+        // Save changes
+        const requestUpdate = cursor.update(json)
+        requestUpdate.onerror = function (event) {
+          // Do something with the error
+        }
+        requestUpdate.onsuccess = function (event) {
+          // Success - the data is updated!
+          console.log('successfully added "' + collectionName + '"')
+        }
+      }
+    }
+  })
+
+  document.querySelector('.collection-area').append(newCollection)
+}
+
+// function getLogInfoAsJSON (cb) {
```

Comment:
Would it be possible to delete this commented block of code? It's my fault for committing it earlier this week (did it for testing purposes before the recent `IndexedDBWrapper.js` changes)

Reply:
Okay it's gone. That reminds me, I need to clean up my code with comments.

Reply:
Same here 😀 we'll assign a bunch of 'spring cleaning' towards the end of this sprint if people have the time